PickMeUp/PickMeUp/viewControllers/DM Features/PMConversationViewController.m

```diff
@@ -24,20 +29,27 @@ @implementation PMConversationViewController
 static const NSString *const kConvoIdKey = @"convoId";
 static const NSString *const kCellIdentifier = @"DMCell";
 static const NSString *const kPostingErrString = @"error on Post request";
+static const NSString *const kLiveQueryURL = @"wss://pickmeup.b4a.io";
 
 - (void)viewDidLoad {
     [super viewDidLoad];
-    self.recieverName.text = self.receiver.username;
+    self.receiverName.text = self.receiver.username;
     self.tableView.dataSource = self;
     self.tableView.delegate = self;
     self.tableView.rowHeight = UITableViewAutomaticDimension;
     self.messageToSend.delegate = self;
-    [NSTimer scheduledTimerWithTimeInterval:5 target:self selector:@selector(onTimer) userInfo:nil repeats:true];
-    [self runGetQuery];
-}
-
-- (void)onTimer {
-    [self runGetQuery];
+    
+    [self _createLiveQueryObj];
+    self.msgQuery = [PFQuery queryWithClassName:kDirectMessageClassName];
+    [self.msgQuery whereKey:kConvoIdKey equalTo:self.convo.objectId];
+    self.subscription = [self.liveQueryClient subscribeToQuery:self.msgQuery];
+    __weak typeof(self) weakSelf = self;
+    [self.subscription addCreateHandler:^(PFQuery<PFObject *> * _Nonnull query, PFObject * _Nonnull object) {
+        __strong typeof(self) strongSelf = weakSelf;
+        [self.arrayOfDMs addObject:(PMDirectMessage *)object];
```

Comment:
Let's move the array change into dispatch main block as well. In general we want to avoid modifying objects from multiple threads, because this can lead to multiple issues: memory corruption, race conditions, etc.